spec/integration/belongs_to_association_spec.rb

```diff
@@ -90,35 +93,61 @@ class SimpleObject < ActiveFedora::Base
         belongs_to :complex_collection, predicate: ActiveFedora::RDF::Fcrepo::RelsExt.isPartOf, class_name: 'ComplexCollection'
       end
 
-      class ComplexObject < SimpleObject
+      #NOTE: As Active Fedora does alphabetical ordering of RDF datastreams automatically, the "Z" is to insure this is stored
```

Comment:
Beyond just that is how it is behaving when you persist the data (ie. if you have two content models on an object, it will always find the alphabetically first one regardless of write order), I'm not exactly sure where it is occurring in this version of the codebase.

In the past, RDF::RDFXML::Writer was used when serializing XML that would have this behavior as its default. I can see that the library that includes that is still in the Gemfile (s.add_dependency "rdf-rdfxml", '~> 1.1.0') but do not see its Writer functionality being used from a codebase search. 

So while it is loading these has_model relationships in alphabetical order, I'd need to dig to see where that is occurring now. Regardless: one can remove the note if wanted (+1 to it). Even if the assertion is incorrect, RDF is not guaranteed to be in any order and the implementation detail is a coincidence. This change is still recommended as though as the only reason several tests failed to fail is because "ComplexObject" came before "SimpleObject" in the alphabet and the current behavior was that automatic re-ordering. You can make just only the naming change in the current branch (before merging this) to see that the tests then would have caught the error. 

There are a few additional corrections and tests added now that will fail in the case the RDF write and read are truly random. But a simple change to help ensure some other tests work in the current implementation of ActiveFedora.